### PR TITLE
Android SDK 12 upgrade

### PR DESCRIPTION
**Pull request by JonasWho**

### CODE REVIEW

File: android/src/main/AndroidManifest.xml [hunk 1]
```diff
@@ -4,6 +4,8 @@
     <uses-permission android:name="android.permission.INTERNET" />
     <!-- Ranging/monitoring beacons -->
     <uses-permission android:name="android.permission.BLUETOOTH"/>
```
Comment (martinpoulsen) [inline android/src/main/AndroidManifest.xml:6]: Specify `android:maxSdkVersion="30"` for this permission as it is not needed for Android 12+. Examples here:
https://developer.android.com/guide/topics/connectivity/bluetooth/permissions#declare

File: android/src/main/AndroidManifest.xml [hunk 2]
```diff
@@ -4,6 +4,8 @@
...
+    <uses-permission android:name="android.permission.BLUETOOTH_SCAN"/>
+    <uses-permission android:name="android.permission.BLUETOOTH_CONNECT"/>
     <!-- Connecting to beacons -->
     <uses-permission android:name="android.permission.BLUETOOTH_ADMIN"/>
```
Comment (martinpoulsen) [inline android/src/main/AndroidManifest.xml:10]: Also specify `android:maxSdkVersion="30"` here.

File: android/src/main/java/com/artirigo/kontaktio/BeaconProximityManager.java [hunk 3]
```diff
@@ -38,6 +41,13 @@ ProximityManager init(ReadableArray beaconTypes) throws Exception {
         // create manager
         proximityManager = ProximityManagerFactory.create(reactAppContext);
 
+        // this forces the new Anchor 2 beacons to include `uniqueId` with beacon ranging data
+        String proximityUuid = String.valueOf(KontaktSDK.DEFAULT_KONTAKT_BEACON_PROXIMITY_UUID);
```
Comment (martinpoulsen) [inline android/src/main/java/com/artirigo/kontaktio/BeaconProximityManager.java:45]: Optimally, this should be possible to configure through the library's API as some consumers of the lib might not use the default kontakt beacon proximity UUID and also some might not want to resolve UUID during ranging for some reason (I believe Kontakt didn't add it by default for some security reason but I'm not certain).

File: android/src/main/java/com/artirigo/kontaktio/BeaconProximityManager.java [hunk 4]
```diff
@@ -38,6 +41,13 @@ ProximityManager init(ReadableArray beaconTypes) throws Exception {
...
+        proximityManager
+                .spaces()
+                .forceResolveRegions(Collections.singletonList(UUID.fromString(proximityUuid)))
+                .forceResolveNamespaces(Collections.singletonList(proximityUuid.refplace("-", "")));
```
Comment (martinpoulsen) [inline android/src/main/java/com/artirigo/kontaktio/BeaconProximityManager.java:49]: `refplace` => `replace`

File: README.md [hunk 5]
```diff
@@ -164,14 +166,22 @@ export default App;
 
     - Android:
 
-      	```bash
+          ```bash
+
       $ react-native run-android
+
+      ```
```
Comment (martinpoulsen) [inline README.md:173]: looks like these should be removed as the end of the code formatting is already defined l. 175

File: README.md [hunk 6]
```diff
@@ -164,14 +166,22 @@ export default App;
...
       ```
 
     - iOS
 
-      	```bash
+          ```bash
+
       $ react-native run-ios
+
+      ```
```
Comment (martinpoulsen) [inline README.md:183]: looks like these should be removed as the end of the code formatting is already defined l. 185

**Review by martinpoulsen — COMMENTED** [review]
Great to get the lib upgraded to support Android 12 target SDK.

**Review by martinpoulsen — APPROVED** [review]
Looks great!

File: CHANGELOG.md [hunk 7]
```diff
@@ -6,6 +6,10 @@ _The format is based on [Keep a Changelog](http://keepachangelog.com/) and this
 
 ## [unreleased]
```
Comment (andrekovac) [inline CHANGELOG.md:7]: Please update the CHANGELOG to represent the current version `4.0.0`

**Pull request merged**